### PR TITLE
style: simplify statements for readability

### LINKED ISSUE #3142: Some statements can be simplified for readability
We can replace lines like

```rust
(len, format!("http://{}/{}", authority, letter))
```

with

```rust
(len, format!("http://{authority}/{letter}"))
```

to improve code readability and maintainability.

### PR DESCRIPTION
**Pull request by hamirmahal**
Fixes #3142

## Motivation

We can simplify some statements, like

```rust
(len, format!("http://{}/{}", authority, letter))
```

to improve code readability.

## Solution

We replace lines like

```rust
(len, format!("http://{}/{}", authority, letter))
```

with

```rust
(len, format!("http://{authority}/{letter}"))
```

### CODE REVIEW

**Review by hamirmahal — COMMENTED** [review]
This shouldn't cause any user-facing changes.

The goal of this pull request is to improve code readability and maintainability.

**Review by kaffarell — APPROVED** [review]
LGTM! Maybe reword the first commit message though – something like "examples: inline variables" would be better IMO.

**Comment by hamirmahal** [comment]
Sure! I'll do that right now.

**Review by kaffarell — APPROVED** [review]
LGTM

**Comment by hamirmahal** [comment]
I just fixed a merge conflict. Are there any other changes I should make?

**Comment by hamirmahal** [comment]
Is [74eeff3](https://github.com/tokio-rs/tracing/pull/3143/commits/74eeff398a1bf2098e7de5e6d0db8c360713c7ac) what you have in mind?

**Review by joshka — APPROVED** [review]

**Comment by joshka** [comment]
LGTM

**Comment by hamirmahal** [comment]
If I fix merge conflicts again, would we be able to merge this?

**Comment by joshka** [comment]
(setting expectations), I'm just a random guy, not a maintainer of tracing, so I don't have merge privileges or anything like that. Perhaps mention it on the discord to get some feedback.

I suspect this could make the backporting efforts more difficult. The master branch gets regularly backported to the 0.1.x branch with changes that affect 0.1 and 0.2. David would probably know this the best.

Something that is possibly a non-obvious thing is that rebasing a larger PR like this can be problematic. It makes it difficult to track (in GitHub at least) which parts have been reviewed and which have not, which means that ensuring that there's no subtle "jia tan xz" type bugs introduced during a rebase is a tediously long process. I reviewed the changes before the force push, not those after.

**Comment by hamirmahal** [comment]
@davidbarsky are you open to this change? I'd like to avoid fixing merge conflicts a third time, if possible.
